data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/OpenSearchSink.java

```diff
@@ -479,11 +479,11 @@ private SerializedJson getDocument(final Event event) {
     if (routingField != null) {
         routingValue = event.get(routingField, String.class);
     } else if (routing != null) {
-        if (expressionEvaluator.isValidFormatExpression(routing)) {
-            routingValue = event.formatString(routing, expressionEvaluator);
-        } else {
-            routingValue = event.get(routing, String.class);
-        }
+      try {
+        routingValue = event.formatString(routing, expressionEvaluator);
+      } catch (final ExpressionEvaluationException | EventKeyNotFoundException e) {
+        LOG.error("Unable to construct routing with format {}, the document_id will be generated by OpenSearch", routing, e);
```

Comment:
Copy past error for `document_id` here. However, the `routing` is different from document id a bit, as it defaults to being the same value as `document_id` (https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-routing-field.html). So we should change this to

```
 LOG.error("Unable to construct routing with format {}, the routing will default to being the same as the document_id", routing, e);
```